### PR TITLE
support cleanPodPolicy is Running, same as tf operator

### PR DESCRIPTION
**Pull request by jiaqianjing**

### CODE REVIEW

**Comment by gaocegege** [comment]
/ok-to-test

/assign @johnugeorge 

Can you please explain more about why we need the PR?

**Comment by jiaqianjing** [comment]
> /ok-to-test
> 
> /assign @johnugeorge
> 
> Can you please explain more about why we need the PR?

1. Consistent experience with TF operator. We can set cleanPodPolicy is "Running", and do the same thing.
2. When I submit a distributed job, when one of the nodes fails, the whole job is in the completed state, but some instances have been completed and some instances are still running. We should clean up thoese "Running" pod and release resource.
At the same time, we can view the log of failed pods.

**Comment by gaocegege** [comment]
SGTM

Can you add corresponding test for the new feature?

**Comment by jiaqianjing** [comment]
> Can you add corresponding test for the new feature?

test_01

**Comment by jiaqianjing** [comment]
PTAL @jinchihe  @andreyvelich

File: pkg/controller.v1/pytorch/job.go [hunk 1]
```diff
@@ -155,13 +155,16 @@ func (pc *PyTorchController) deletePodsAndServices(job *pyv1.PyTorchJob, pods []
...
-	if *job.Spec.CleanPodPolicy == common.CleanPodPolicyNone ||
-		*job.Spec.CleanPodPolicy == common.CleanPodPolicyRunning {
+	// Delete nothing when the cleanPodPolicy is None.
+	if *job.Spec.CleanPodPolicy == common.CleanPodPolicyNone {
 		return nil
 	}
 
 	for _, pod := range pods {
+		// Just delete running pod when the cleanPodPolicy is Running
+		if *job.Spec.CleanPodPolicy == common.CleanPodPolicyRunning && pod.Status.Phase != v1.PodRunning {
```
Comment (andreyvelich) [inline pkg/controller.v1/pytorch/job.go:165]: Maybe we could add new test case, like this: https://github.com/kubeflow/pytorch-operator/blob/76a240f180da9fcddfcc87bdd8a18b0a4fe33da3/pkg/controller.v1/pytorch/job_test.go#L221-L240 when `*job.Spec.CleanPodPolicy == common.CleanPodPolicyRunning && pod.Status.Phase != v1.PodRunning` ?

Comment (jiaqianjing) [reply pkg/controller.v1/pytorch/job.go:165]: 3q, I've recommitted

**Comment by gaocegege** [comment]
Thanks for your contribution! :tada: :+1:

File: pkg/controller.v1/pytorch/job_test.go [hunk 2]
```diff
@@ -238,6 +238,46 @@ func TestDeletePodsAndServices(t *testing.T) {
...
+
+			activeWorkerServices: 4,
+			activeMasterServices: 1,
+
+			expectedPodDeletions: 5,
+			expectedServiceDeletions: 1,
+		},
+		testCase{
+			description: "4 workers and 1 master succeeded, policy is running",
+			job: 		 testutil.NewPyTorchJobWithCleanPolicy(1, 4, common.CleanPodPolicyRunning),
```
Comment (gaocegege) [inline pkg/controller.v1/pytorch/job_test.go:263]: Please use gofmt to format the code.

Comment (jiaqianjing) [reply pkg/controller.v1/pytorch/job_test.go]: execute command: "gofmt -l -w  ./pkg/controller.v1/pytorch/job_test.go"

**Comment by gaocegege** [comment]
/approve

@johnugeorge @andreyvelich They need this feature urgently thus I am merging it. If you have any comment feel free to leave. We can fix in another PR.

**Pull request merged**